src/metabase/util/malli/fn.clj

```diff
@@ -143,20 +146,36 @@
   use [[metabase.util.malli/disable-enforcement]] to bind this only in Clojure code."
   true)
 
+(defn- fixup-stacktrace
+  "This function removes stack trace elements that came from this namespace. When we throw validation errors, they
+  shouldn't originate from *this* namespace, they should appear to be thrown from the instrumented function itself."
+  [^Exception e]
+  (let [trace (.getStackTrace e)
+        fixed-trace (into-array StackTraceElement
+                                (drop-while
+                                 #(str/starts-with? (.getClassName %)
+                                                    ;; this is... hacky, but it works.
+                                                    (namespace ::x))
```

Comment:
You called out that it's hacky (it's not so bad is it? :) ).

Anyway what do you think of `(str *ns*)`?

Reply:
That was actually my initial impulse! But sadly, it doesn't seem like `*ns*` is bound to "the namespace I'm currently in," exactly. For an example, I defined a function `test-it` in a random namespace, `metabase.server.middleware.session`:

```
(defn test-it []
  (str *ns*))
```

Then called it from the `user` namespace:

```
(metabase.server.middleware.session/test-it)
```

The result was `user`, not `metabase.server.middleware.session`. 😢